src/server.c

```diff
@@ -635,7 +635,7 @@ const char *strChildType(int type) {
 /* Return true if there are active children processes doing RDB saving,
  * AOF rewriting, or some side process spawned by a loaded module. */
 int hasActiveChildProcess(void) {
-    return server.child_pid != -1;
+    return server.child_pid != -1 && server.child_type != CHILD_TYPE_NONE;
```

Comment:
the test failed because in here, when loading a config file, the server.child_pid is 0 (we reset it to -f after the config loading)

Reply:
the other options is:
1. using `reading_config_file` and skip it.
2. init server.child_pid before the loading.